### PR TITLE
Decorate a Class

### PR DESCRIPTION
**Pull request by Cosmosha**
### The following requirement and features was implemented;

- Gitflow
- Linters
- Add Nameable Class
- Add Base Decorator Class
- Add Capitalize Decorator Class
- Add Trimmer Decorator Class
- Check to decorate Person

### CODE REVIEW

**Review by rloterh — COMMENTED** [review]
Hi Cosmos, 

While you made a great effort in this project, unfortunately, I cannot proceed to review your code.

## Invalid Code Review Request

Kindly ensure that your ReadMe is professionally documented; The _Key Features_ section of your ReadMe should be present and includes 1-3 features of your app. 

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: trimmer_decorator.rb [hunk 1]
```diff
@@ -0,0 +1,7 @@
+require './base_decorator'
+
+class TrimmerDecorator < BaseDecorator
+  def correct_name
+    super[0.9]
+  end
```
Comment (codepantha) [inline trimmer_decorator.rb]: - [ ] The correct name method here doesn't return the expected value; It only gives us the first character of the name as you can see in the attached image below 👇 
I am sure you meant to do `super[0..9]`?
image

File: person.rb [hunk 2]
```diff
@@ -0,0 +1,29 @@
+require './nameable'
```
Comment (codepantha) [inline person.rb:1]: - [ ] [OPTIONAL] Though require can be used to both execute gems and external dependencies, the preferable method to load relative paths is `require_relative`. Therefore, I would encourage you to replace `require` with `require_relative` whenever you are loading relative paths and everywhere it occurs in your application in order to keep in line with best practices 😎

**Review by codepantha — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED ♻️ 

## Hi @Cosmosha 👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights 🎯 
✔️ Descriptive PR
✔️ Implemented decorators 📌 
✔️ Linters are passing

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- n/a

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: trimmer_decorator.rb (hunk 1 above)
Comment (Cosmosha) [reply trimmer_decorator.rb]: Well note, Thank you so much for the feedback

**Review by m05e5 — APPROVED** [review]
**Hi @Cosmosha 
You did great. Your project is complete!  There is nothing else to say other than... it's time to merge:shipit: 
Congratulations! 🎉 
CleverScornfulBuffalo-size_restricted

## Highlights

✔️ Great job implementing the requested changes

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

_**m05e5**_

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._**

**Pull request merged**